include/treelite/tree.h

```diff
@@ -474,12 +488,25 @@ class Model {
   /*! \brief disable copy; use default move */
   Model() = default;
   virtual ~Model() = default;
-  inline static std::unique_ptr<Model> Create();
   Model(const Model&) = delete;
   Model& operator=(const Model&) = delete;
   Model(Model&&) = default;
   Model& operator=(Model&&) = default;
 
+  template <typename ThresholdType, typename LeafOutputType>
+  inline static std::unique_ptr<Model> Create();
```

Comment:
Consider moving this function into `ModelImpl`, as it depends on `ThresholdType` and `LeafOutputType`.

Reply:
It is actually not possible to move the function, since it needs access the base class's members `threshold_type_` and `leaf_output_type_`. The members were moved to the base class per your previous suggestion.